cmd/connections.go

```diff
@@ -465,3 +475,636 @@ func terminateConnectionByUser(options connectionOptions) error {
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.IntVar(&options.bin_count, "bin_count", 50, "how many bins from training history")
+	flags.BoolVarP(&options.logs, "logs", "l", false, "add pandas and tensorflow log prints")
+
+	cmd.AddCommand(uebaDatasetCreateCmd())
+	cmd.AddCommand(uebaDatasetShowCmd())
+	cmd.AddCommand(uebaDatasetUpdateCmd())
+	cmd.AddCommand(uebaDatasetDeleteCmd())
+	cmd.AddCommand(uebaDatasetTrainCmd())
+	cmd.AddCommand(uebaConnectionCountsCmd())
+
+	return cmd
+}
+
+func uebaDatasets(options uebaOptions) error {
+	api := connectionmanager.New(curl())
+
+	datasets, err := api.UebaDatasets(options.logs, options.bin_count)
+	if err != nil {
+		return err
+	}
+
+	return stdout(datasets)
+}
+
+//
+//
+func uebaDatasetCreateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create new dataset definition.",
+		Long:  `Create new dataset definition.`,
+		Example: `
+	privx-cli ueba datasets create [access flags] JSON-FILE
+			`,
+		Args:         cobra.ExactArgs(1),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return uebaDatasetCreate(args)
+		},
+	}
+
+	return cmd
+}
+
+func uebaDatasetCreate(args []string) error {
+	var datasetBodyParam connectionmanager.DatasetBodyParam
+	api := connectionmanager.New(curl())
+
+	err := decodeJSON(args[0], &datasetBodyParam)
+	if err != nil {
+		return err
+	}
+
+	datasetID, err := api.CreateUebaDataset(datasetBodyParam)
+	if err != nil {
+		return err
+	}
+
+	return stdout(datasetID)
+}
+
+//
+//
+func uebaDatasetShowCmd() *cobra.Command {
+	options := uebaOptions{}
+	cmd := &cobra.Command{
+		Use:   "show",
+		Short: "Get dataset by ID",
+		Long:  `Get dataset by ID, possibility to filter training history`,
+		Example: `
+	privx-cli ueba datasets show [access flags] --id <DATASET-ID>
+	privx-cli ueba datasets show [access flags] --id <DATASET-ID> --logs false --bin_count 50
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return uebaDatasetShow(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.datasetID, "id", "", "dataset ID")
+	flags.IntVar(&options.bin_count, "bin_count", 50, "how many bins from training history")
+	flags.BoolVarP(&options.logs, "logs", "l", false, "add pandas and tensorflow log prints")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func uebaDatasetShow(options uebaOptions) error {
+	api := connectionmanager.New(curl())
+
+	dataset, err := api.UebaDataset(options.logs, options.bin_count, options.datasetID)
+	if err != nil {
+		return err
+	}
+
+	return stdout(dataset)
+}
+
+//
+//
+func uebaDatasetUpdateCmd() *cobra.Command {
+	options := uebaOptions{}
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update dataset",
+		Long:  `Update dataset. Note this will cause backend to empty training history and delete trained weights in ueba machine.`,
+		Example: `
+	privx-cli ueba datasets update [access flags] JSON-FILE --id <DATASET-ID>
```

Comment:
all flags should precede positional argument

Reply:
corrected.

In the meantime, I see quite a few commands in other parts of the code (for eg, [here](https://github.com/SSHcom/privx-cli/blob/master/cmd/hosts.go#L224)), where the sequence of flag and positional argument does not seem to be followed. Maybe we need to correct those as well.

If you give me a signal, I can quickly correct those in other places as well in another MR.